src/internal/database/tenant.ts

```diff
@@ -250,11 +262,169 @@ export async function listenForTenantUpdate(pubSub: PubSubAdapter): Promise<void
     tenantConfigCache.delete(cacheKey)
   })
 
+  await pubSub.subscribe(TENANTS_JWKS_UPDATE_CHANNEL, (cacheKey) => {
+    tenantJwksConfigCache.delete(cacheKey)
+  })
+
   await pubSub.subscribe(TENANTS_S3_CREDENTIALS_UPDATE_CHANNEL, (cacheKey) => {
     tenantS3CredentialsCache.delete(cacheKey)
   })
 }
 
+/**
+ * Gets a list of all tenants that do not have a signing url associated
+ */
+export async function* listTenantsMissingUrlSigningJwk(
+  signal: AbortSignal,
+  batchSize = 200
+): AsyncGenerator<string[]> {
+  let lastCursor = 0
+
+  while (!signal.aborted) {
+    const data = await multitenantKnex('tenants')
+      .select('id', 'cursor_id')
+      .where('cursor_id', '>', lastCursor)
+      .whereNotExists(function () {
```

Comment:
You don't need `whereNotExists` you can simply chain the conditions on the main query:

```ts
 .andWhere('tenants_jwks.kind', JWK_KIND_STORAGE_URL_SIGNING)
 .andWhere('tenants_jwks.active', true)

```